### PR TITLE
Use constructable with NetBox dynamic inventory

### PR DESCRIPTION
**Pull request by remyleone**
##### SUMMARY

As mention by @s-hertel in https://github.com/ansible/ansible/pull/45741#discussion_r218164978 
`Constructable` can be used to simplify inventory.

The goal of this PR is to refactor the code of the NetBox inventory using Constructable to avoid adding new functions every time a new attribute for an host is added.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

- lib/ansible/plugins/inventory/netbox.py

##### ANSIBLE VERSION

```
ansible 2.8.0.dev0 (netbox_constructable 5415677520) last updated 2018/09/20 15:10:57 (GMT +200)
  config file = None
  configured module search path = [u'/Users/sieben/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/sieben/workspace/ansible/lib/ansible
  executable location = /Users/sieben/workspace/ansible/bin/ansible
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

### CODE REVIEW

**Review by s-hertel — APPROVED** [review]
LGTM, a simple example or two would be nice too

**Comment by sieben** [comment]
@s-hertel I've tried to make it work with the following example:

```
plugin: netbox
api_endpoint: http://localhost:8000
query_filters:
  - name: switch_7714
variables:
  foo: last_updated
  bar: display_name
```

But no variable `foo` or `bar` shows up. Am I using this correctly?

**Comment by remyleone** [comment]
@s-hertel I've added an example in the documentation. I've also tested and it works. Could we merge it?

**Comment by remyleone** [comment]
@gundalow Could we merge it?

File: lib/ansible/plugins/inventory/netbox.py [hunk 1]
```diff
@@ -50,10 +52,15 @@
...
             description: List of parameters passed to the query string (Multiple values may be separated by commas)
             type: list
+            default: []
         timeout:
             description: Timeout for Netbox requests in seconds
             type: int
             default: 60
+        compose:
+            description: List of custom ansible host vars to create from the device object fetched from NetBox
+            default: {}
```
Comment (gundalow) [inline lib/ansible/plugins/inventory/netbox.py:63]: Do we need `version_added` here?

Comment (remyleone) [reply lib/ansible/plugins/inventory/netbox.py:63]: Netbox is not present in 2.7 so I guess that all of it will be present for 2.8

**Comment by remyleone** [comment]
@gundalow Could we merge it?

**Comment by remyleone** [comment]
@s-hertel Could we merge it?

**Comment by remyleone** [comment]
@gundalow Could we merge it?

**Comment by remyleone** [comment]
@Anthony25 could you review?

**Comment by aruhier** [comment]
LGTM

**Comment by gundalow** [comment]
Merged into `devel` for release in Ansible 2.8

**Pull request merged**